sphinx_gherkindoc/writer.py

```diff
@@ -232,14 +236,18 @@ def table(table: behave.model.Table, inline: bool = False) -> None:
     feature = behave.parser.parse_file(source_path)
     section(1, feature)
     description(feature.description)
-    if feature.background:
+    if feature.background and not integrate_background:
         section(2, feature.background)
         steps(feature.background.steps)
+        output_file.blank_line()
     for scenario in feature.scenarios:
         section(2, scenario)
         tags(scenario.tags, feature)
         description(scenario.description)
+        if integrate_background:
+            steps(feature.background.steps, step_format=background_step_format)
         steps(scenario.steps)
+        output_file.blank_line()
```

Comment:
Since there may be more than one call to `steps` in a single scenario, I had to extract the blank line until all of the steps are done for that section.